src/key.rs

```diff
@@ -485,6 +504,12 @@ mod test {
         assert!(compressed.is_ok());
     }
 
+    #[test]
+    fn pubkey_as_slice() {
+        let pubkey = PublicKey::from_slice(&[4, 54, 57, 149, 239, 162, 148, 175, 246, 254, 239, 75, 154, 152, 10, 82, 234, 224, 85, 220, 40, 100, 57, 121, 30, 162, 94, 156, 135, 67, 74, 49, 179, 57, 236, 53, 162, 124, 149, 144, 168, 77, 74, 30, 72, 211, 229, 110, 111, 55, 96, 193, 86, 227, 183, 152, 195, 155, 51, 247, 123, 113, 60, 228, 188]).unwrap();
+        assert_eq!(pubkey.as_slice(), &[179, 49, 74, 67, 135, 156, 94, 162, 30, 121, 57, 100, 40, 220, 85, 224, 234, 82, 10, 152, 154, 75, 239, 254, 246, 175, 148, 162, 239, 149, 57, 54, 188, 228, 60, 113, 123, 247, 51, 155, 195, 152, 183, 227, 86, 193, 96, 55, 111, 110, 229, 211, 72, 30, 74, 77, 168, 144, 149, 124, 162, 53, 236, 57][..]);
+    }
```

Comment:
Can't understand why `from_slice` and `as_slice` byte sequences are different...

Reply:
That's because the internal array of PublicKey isn't a valid anything.
it's an opaque blob of data for upstream, it can change per OS/system/version, and should never be accessed and/or saved.

that's why we have serialize+parse methods.

Reply:
@elichai but this internal array which is `isn't a valid anything` is exposed via `as_ptr` and `as_mut_ptr` as well as internal representation of `PrivateKey`, which is "valid" by the way. 
What's the reason to store something `isn't a valid anything` inside and not to store "serialised" representation which will be used everywhere anyway?  
This may confuse people because every time to get "valid" `PublicKey` they must call `serialise` method which isn't zero-cost or store it in some custom buffer, but why?

Reply:
well that's a tradeoff, when should the cost of `serialize` be paid? oh and don't forget that then any other operation on PublicKey will require the cost of `parse` again. (FYI it's not zero-cost but it's pretty damn cheap).

About the pointer, these are meant for FFI reasons, because there are users of this library who also talk to the FFI themselves so they need a raw pointer to the internal array.